src/pipert2/core/handlers/message_handlers/queue_handler.py

```diff
@@ -0,0 +1,85 @@
+from ..message_handler import MessageHandler
+from multiprocessing import Queue
+from queue import Full, Empty
+from pipert2.core.base.message import Message
+
+
+class QueueHandler(MessageHandler):
+    """The queue message handler implements the functionality needed to get and put messages using multiprocessing
+    queues.
+
+    Args:
+        input_queue: The that will be used in order to get messages.
+        output_queue: The that will be used in order to push messages.
+        blocking: If the queues will behave as blocking behavior detailed in each function or not.
+        timeout: How long the queues will wait in seconds if blocking is true.
+
+    """
+
+    def __init__(self, input_queue: Queue, output_queue: Queue, blocking=True, timeout=1):
+        super().__init__()
+        self.input_queue = input_queue
+        self.output_queue = output_queue
+        self.blocking = blocking
+        self.timeout = timeout
+
+    def _get(self):
+        """Get a message from the input queue.
+        If blocking is true, wait the set timeout for a message to arrive,
+        otherwise do nothing if the queue is empty.
+
+        Returns:
+            Return the message that came from the queue or None if the queue was empty.
+
+        """
+        message = None
+
+        try:
+            message = self.input_queue.get(block=self.blocking, timeout=self.timeout)
+        except Empty:
+            print("The queue is empty")  # TODO: Replace with log
+
+        return message
+
+    def _put(self, message: Message):
```

Comment:
What if we want to have a blocking time and if the queue doesn't available after the time, we force push ?

Reply:
No, we give up on the message and log the error